### PR TITLE
FNEG and addition FP value mutations

### PR DESCRIPTION
**Pull request by isaacdbeans**
Added a new helper for unary operators including FNeg
Added support for random double and float values.
Added support for extreme double and float values.
Added support for used double and float values.

Also CCing John in this pull at his request
@regehr

### CODE REVIEW

File: tools/mutator-utils/mutator.cpp [hunk 1]
```diff
@@ -466,12 +482,20 @@ void FunctionMutator::fixAllValues(llvm::SmallVector<llvm::Value *> &vals) {
 llvm::Value *FunctionMutator::getRandomConstant(llvm::Type *ty) {
   if (ty->isIntegerTy()) {
     return llvm::ConstantInt::get(
-        ty->getContext(),
```
Comment (regehr) [inline tools/mutator-utils/mutator.cpp:469]: please try to avoid whitespace changes in pull requests. if we want to reformat the whitespace, that should be done in a totally separate PR

File: tools/mutator-utils/mutator.h [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #include "llvm/ADT/StringSet.h"
 #include "llvm/Analysis/TargetLibraryInfo.h"
 #include "llvm/Bitcode/BitcodeReader.h"
+#include "llvm/Config/llvm-config.h.cmake"
```
Comment (regehr) [inline tools/mutator-utils/mutator.h]: is this change necessary?

File: tools/mutator-utils/mutator_helper.cpp [hunk 3]
```diff
@@ -1131,4 +1133,53 @@ void ResizeIntegerHelper::debug() {
   llvm::errs() << "integer resized\n";
   mutator->iitInTmp->getParent()->print(llvm::errs());
   llvm::errs() << "\n";
-}
\ No newline at end of file
+}
+
+void UnaryInstHelper::init() {
```
Comment (regehr) [inline tools/mutator-utils/mutator_helper.cpp]: do you need a separate init() method? it would be better to do this in the constructor, or in just a regular static initializer

File: tools/mutator-utils/mutator_helper.h [hunk 4]
```diff
@@ -342,3 +342,21 @@ class ResizeIntegerHelper : public MutationHelper {
...
+
+class UnaryInstHelper : public MutationHelper {
+  bool updated;
+public:
+ UnaryInstHelper(std::shared_ptr<FunctionMutator> mutator)
+      : MutationHelper(mutator), updated(false) {};
+  virtual void init() override;
+  virtual void reset() override { 
+    updated = false; 
+    }
```
Comment (regehr) [inline tools/mutator-utils/mutator_helper.h]: please be careful with indentation-- it's best to your `clang-format -i` on your code before issuing a pull request

File: tools/mutator-utils/mutator.cpp (hunk 1 above)
Comment (Hatsunespica) [reply tools/mutator-utils/mutator.cpp:469]: I agree. I saw lots of whitespace changes.
You can do `git clang-format` to only format those changed lines after `git add` @isaacdbeans

File: tools/mutator-utils/mutator_helper.cpp [hunk 5]
```diff
@@ -1,5 +1,7 @@
 #include "mutator_helper.h"
 #include "mutator.h"
+#include <iostream>
+#include <random>
```
Comment (Hatsunespica) [inline tools/mutator-utils/mutator_helper.cpp]: I don't think we need both `include` here?
random is included in `util.cpp`, and I don't see why we need an `iostream` here

File: tools/mutator-utils/mutator.cpp (hunk 1 above)
Comment (isaacdbeans) [reply tools/mutator-utils/mutator.cpp:469]: good to know! i ended up clang-formating everything

**Comment by isaacdbeans** [comment]
Still getting used to making sure all the changes are made before committing haha 😅
